packages/teams-js/test/public/app.spec.ts

```diff
@@ -11,585 +11,1213 @@ import {
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, null],
+          },
+        } as DOMMessageEvent);
+        await expect(promise).rejects.toThrowError('Received runtime config is invalid');
+      });
+
+      it('app.initialize should not use the teams config as a default if another proper config is given', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = framelessPostMock.findMessageByFunc('initialize');
+        framelessPostMock.respondToMessage({
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, '{"apiVersion":1, "supports":{"mail":{}}}'],
+          },
+        } as DOMMessageEvent);
+        await initPromise;
+
+        expect(runtime).not.toEqual(teamsRuntimeConfig);
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+      });
+
+      it('app.initialize should assign clientSupportedSDKVersion correctly when a proper runtime config is given', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = framelessPostMock.findMessageByFunc('initialize');
+        framelessPostMock.respondToMessage({
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, '{"apiVersion":1, "supports":{"mail":{}}}', '1.0.0'],
+          },
+        } as DOMMessageEvent);
+        await initPromise;
+
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+        expect(GlobalVars.clientSupportedSDKVersion).toBe('1.0.0');
+      });
+
+      it('app.initialize should initialize with clientSupportedSDKVersion and runtimeConfig arguments flipped', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = framelessPostMock.findMessageByFunc('initialize');
+        framelessPostMock.respondToMessage({
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, '1.0.0', '{"apiVersion":1, "supports":{"mail":{}}}'],
+          },
+        } as DOMMessageEvent);
+        await initPromise;
+
+        expect(runtime).toEqual({ apiVersion: 1, supports: { mail: {} } });
+        expect(GlobalVars.clientSupportedSDKVersion).toBe('1.0.0');
+      });
+
+      it('app.initialize should initialize with teams config when an invalid runtimeConfig is given, with arguments flipped', async () => {
+        const initPromise = app.initialize();
+
+        const initMessage = framelessPostMock.findMessageByFunc('initialize');
+        framelessPostMock.respondToMessage({
+          data: {
+            id: initMessage.id,
+            args: [FrameContexts.content, HostClientType.web, '1.6.0', 'nonJSONStr'],
+          },
+        } as DOMMessageEvent);
+        await initPromise;
+
+        expect(runtime).toEqual(teamsRuntimeConfig);
+      });
     });
```

Comment:
I think the same comments in the framed tests apply to the frameless ones for `app.initialize` test cases.